utils/cache_manager.py

```
@@ -1,20 +1,131 @@
+            checksum = generate_checksum(value)  # Generate checksum using the same serialized value
+            key = sanitize_cache_key(key)
+
+            # Debugging: Print checksum and serialized value
+            #print(f"Generated checksum during set: {checksum}")
+            #print(f"Serialized value during set: {serialized_value}")
+
+            # Store value and checksum together
+            self.redis_client.setex(
+                key, timeout, json.dumps({"value": value, "checksum": checksum}, cls=jencoder)
+            )
+        except:
```

Ruff (0.8.0)

Do not use bare `except`

(E722)